angular/lib/src/lib/slider/slider.component.ts

```diff
@@ -0,0 +1,539 @@
+			}
+
+			.au-slider-progress {
+				background-color: var(--bs-primary, #005eb8);
+				height: 0.25rem;
+				position: absolute;
+
+				&-vertical {
+					transform: rotate(90deg);
+				}
+			}
+
+			.au-slider-label {
+				margin-top: -1.9rem;
+				position: absolute;
+
+				&-min {
+					left: 0;
+				}
+
+				&-max {
+					right: 0;
+				}
+
+				&-now {
+					transform: translateX(-50%);
+				}
+			}
+
+			.au-slider-label-vertical {
+				margin-left: 1rem;
+				position: absolute;
+				transform: translateY(-40%);
+				white-space: nowrap;
+
+				&-min {
+					top: 100%;
+				}
+
+				&-max {
+					top: 0%;
+				}
+
+				&-now {
+					transform: translateY(-50%);
+				}
+			}
+
+			.au-slider {
+				display: flex;
+				background-color: var(--bs-secondary-bg, #d3d3d3);
+				position: relative;
+			}
+
+			.au-slider-horizontal {
+				width: 100%;
+				height: 0.25rem;
+				margin-top: 2rem;
+				margin-bottom: 1rem;
+			}
+
+			.au-slider-vertical {
+				height: 100%;
+				width: 0.25rem;
+				margin-right: 2rem;
+				margin-left: 1rem;
+				margin-bottom: 1rem;
+				margin-top: 1rem;
+			}
+
+			.au-slider-progress[disabled],
+			.au-slider-handle[disabled] {
+				background-color: var(--bs-dark-bg-subtle, #b3b3b3);
+				cursor: not-allowed;
+			}
+
+			.au-slider-label[disabled],
+			.au-slider-label-vertical[disabled] {
+				color: var(--bs-dark-bg-subtle, #b3b3b3);
+			}
+
+			au-slider[disabled] {
+				cursor: not-allowed;
+			}
+
+			.au-clickable-slider-area {
+				height: 1.5rem;
+				width: 100%;
+				transform: translateY(-50%);
+			}
+
+			.au-clickable-slider-area-vertical {
+				width: 1.5rem;
+				height: 100%;
+				transform: translateX(-50%);
+				flex-shrink: 0;
+			}
+		`,
+	],
+})
+export class SliderComponent implements OnChanges, AfterViewInit, OnDestroy {
+	/**
+	 * auSlider element reference
+	 */
+	private _elementRef = inject(ElementRef);
+	private _destroyRef = inject(DestroyRef);
+
+	private _resizeObserver = new ResizeObserver(() => {
+		this._widget.patch({
+			sliderDomRect: this._elementRef.nativeElement.getBoundingClientRect(),
+			minLabelDomRect: this._minLabelRef?.nativeElement.getBoundingClientRect(),
+			maxLabelDomRect: this._maxLabelRef?.nativeElement.getBoundingClientRect(),
+		});
+	});
+
+	/**
+	 * Minimum label element reference
+	 */
+	@ViewChild('minLabel')
+	private _minLabelRef: ElementRef;
+	/**
+	 * Maximum label element reference
+	 */
+	@ViewChild('maxLabel')
+	private _maxLabelRef: ElementRef;
+
+	readonly _widget = callWidgetFactory(createSlider, 'slider');
+	/**
+	 * SliderWidget hold the state and actions applicable to the auSliderComponent
+	 */
+	readonly widget = toSlotContextWidget(this._widget);
+	readonly state: Signal<SliderState> = toSignal(this._widget.state$, {requireSync: true});
+	readonly api = this._widget.api;
+
+	/**
+	 * Dom Rectangle of the slider
+	 */
+	@Input('auSliderDomRect') sliderDomRect: DOMRect | undefined;
```

Comment:
Why is is a component input?

Reply:
from my understanding there should be input for every prop, otherwise it doesn't build properly

Reply:
It should not be a prop